File: Animation/app/src/main/java/com/example/animation/MainActivity.java
package com.example.animation;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

import android.app.Dialog;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.ViewGroup;
import android.view.WindowManager;
import android.widget.Button;
import android.widget.Toast;

import com.airbnb.lottie.LottieAnimationView;

public class MainActivity extends AppCompatActivity {
    LottieAnimationView animationView;
    Button button , demo;
    Dialog dialog;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
//        getSupportActionBar().hide();

        demo = findViewById(R.id.demo);
        dialog = new Dialog(MainActivity.this);
        dialog.setContentView(R.layout.dialogbox);
        //dialog.getWindow().setLayout(ViewGroup.LayoutParams.MATCH_PARENT,ViewGroup.LayoutParams.WRAP_CONTENT);

        Button Yes = dialog.findViewById(R.id.yes);
        Button No = dialog.findViewById(R.id.no);

        Yes.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                dialog.dismiss();
            }
        });
        No.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(MainActivity.this, "You click on NO", Toast.LENGTH_SHORT).show();
            }
        });
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                dialog.show();
            }
        });


        getWindow().addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN);
        setContentView(R.layout.activity_main);
        animationView = findViewById(R.id.animation);
        button = findViewById(R.id.demo);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                animationView.setVisibility(View.VISIBLE);
                button.setVisibility(View.INVISIBLE);
                animationView.playAnimation();
                Thread td =new Thread(){
                    public void run()
                    {
                        try {
                            sleep(1000);
                        }
                        catch (Exception e){
                            e.printStackTrace();
                        }
                        finally {
                            Intent intent = new Intent(MainActivity.this,Demo.class);
                            startActivity(intent);
                            finish();
                        }
                    }

                };td.start();
            }
        });
        Intent intent = new Intent(this,ServiceExamp.class);
        ContextCompat.startForegroundService(this,intent);

    }
}

File: MyTictactoe/app/src/main/java/com/example/mytictactoe/MainActivity.kt
package com.example.mytictactoe

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.Toast
import android.widget.Toolbar
import androidx.appcompat.app.ActionBar

class MainActivity : AppCompatActivity() {
    private lateinit var btn1: Button
    private lateinit var btn2: Button
    private lateinit var btn3: Button
    private lateinit var btn4: Button
    private lateinit var btn5: Button
    private lateinit var btn6: Button
    private lateinit var btn7: Button
    private lateinit var btn8: Button
    private lateinit var btn9: Button
    private var flag:Int = 0
    private var count:Int = 0
    private lateinit var b1:String
    private lateinit var b2:String
    private lateinit var b3:String
    private lateinit var b4:String
    private lateinit var b5:String
    private lateinit var b6:String
    private lateinit var b7:String
    private lateinit var b8:String
    private lateinit var b9:String
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        init()
    }
    private fun  init(){


        btn1 = findViewById(R.id.btn1)
        btn2 = findViewById(R.id.btn2)
        btn3 = findViewById(R.id.btn3)
        btn4 = findViewById(R.id.btn4)
        btn5 = findViewById(R.id.btn5)
        btn6 = findViewById(R.id.btn6)
        btn7 = findViewById(R.id.btn7)
        btn8 = findViewById(R.id.btn8)
        btn9 = findViewById(R.id.btn9)
    }

    fun check(v: View){

        val btnCurrent: Button = v as Button


        if(btnCurrent.text.toString() == "") {
            count++

            if (flag == 0) {
                btnCurrent.text = "X"
                flag = 1
            } else {
                btnCurrent.text = "O"
                flag = 0
            }
            if (count > 4) {
                b1 = btn1.text.toString()
                b2 = btn2.text.toString()
                b3 = btn3.text.toString()
                b4 = btn4.text.toString()
                b5 = btn5.text.toString()
                b6 = btn6.text.toString()
                b7 = btn7.text.toString()
                b8 = btn8.text.toString()
                b9 = btn9.text.toString()

                if (b1 == b5 && b5 == b9 && b1!="") {
                    //1
                    Toast.makeText(this, "Winner is: ${this.b5}", Toast.LENGTH_SHORT).show()
                    gameOver()
                }
                else if (b3 == b5 && b5 == b7 && b3 != "") {
                    //2
                    Toast.makeText(this, "Winner is: ${this.b5}", Toast.LENGTH_SHORT).show()
                    gameOver()
                } else if (b1 == b2 && b2 == b3 && b2 != "") {
                    //3
                    Toast.makeText(this, "Winner is: ${this.b2}", Toast.LENGTH_SHORT).show()
                    gameOver()
                } else if (b4 == b5 && b5 == b6 && b4 != "") {
                    //4
                    Toast.makeText(this, "Winner is: ${this.b4}", Toast.LENGTH_SHORT).show()
                    gameOver()
                } else if (b7 == b8 && b8 == b9 && b7 !="") {
                    //5
                    Toast.makeText(this, "Winner is: ${this.b7}", Toast.LENGTH_SHORT).show()
                    gameOver()
                } else if (b3 == b6 && b6 == b9 && b3 != "") {
                    //6
                    Toast.makeText(this, "Winner is: ${this.b3}", Toast.LENGTH_SHORT).show()
                    gameOver()
                } else if (b2 == b5 && b5 == b8 && b2 != "") {
                    //7
                    Toast.makeText(this, "Winner is: ${this.b2}", Toast.LENGTH_SHORT).show()
                    gameOver()
                } else if (b1 == b4 && b4 == b7 && b1 != "") {
                    //8
                    Toast.makeText(this, "Winner is: ${this.b1}", Toast.LENGTH_SHORT).show()
                    gameOver()
                }
                else if(b1!="" && b2!="" && b3!=""
                    && b4!="" && b5!="" && b6!=""
                    &&b7!="" && b8!="" && b9!="") {

                    Toast.makeText(this, "Game is Drawn", Toast.LENGTH_SHORT).show()
                    gameOver()
                }
            }
        }



    }

    private fun gameOver(){
        btn1.text = ""
        btn2.text = ""
        btn3.text = ""
        btn4.text = ""
        btn5.text = ""
        btn6.text = ""
        btn7.text = ""
        btn8.text = ""
        btn9.text = ""
        count = 0
        flag = 0
    }

}
